### PR TITLE
docker: build and publish multiarch images

### PR DESCRIPTION
**Pull request by s-urbaniak**
cc @paulfantom 

Signed-off-by: Sergiusz Urbaniak <sergiusz.urbaniak@gmail.com>

### CODE REVIEW

File: scripts/publish.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+# Create manifest to join all images under one virtual tag
+MANIFEST="docker manifest create -a ${QUAY_PATH}:${VERSION}"
+for arch in ${CPU_ARCHS}; do
+  MANIFEST="${MANIFEST} ${QUAY_PATH}:${VERSION}-${arch}"
+done
+eval "${MANIFEST}"
+
+# Annotate to set which image is build for which CPU architecture
+for arch in ${CPU_ARCHS}; do
+  docker manifest annotate --arch "${arch}" "${QUAY_PATH}:${VERSION}" "${QUAY_PATH}:${VERSION}-${arch}"
```
Comment (s-urbaniak) [inline scripts/publish.sh:35]: @paulfantom I think we also have to:
```
  docker manifest push "${QUAY_PATH}:${VERSION}-${arch}"
```
here, no?

Comment (paulfantom) [reply scripts/publish.sh:35]: It isn't necessary as `docker manifest push` is smart enough to push all images annotated to virtual manifest (the one created with `docker manifest create`).

File: scripts/publish.sh [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+set -e
+# only exit with zero if all commands of the pipeline exit successfully
+set -o pipefail
+# error on unset variables
+set -u
+
+CPU_ARCHS="amd64 arm64 arm ppc64le s390x"
+TAG_COMMIT=$(git rev-list --abbrev-commit --tags --max-count=1)
+COMMIT=$(git rev-parse --short HEAD)
+TAG=$(git describe --abbrev=0 --tags ${TAG_COMMIT} 2>/dev/null || true)
```
Comment (paulfantom) [inline scripts/publish.sh:12]: Why setting `|| true`? Do we expect `git describe` to be able to fail?

File: scripts/publish.sh [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+VERSION="${TAG}"
+
+# if the current commit, does not correspond to a tag, create a verbose version string.
+if [ "${TAG_COMMIT}" != "${COMMIT}" ]; then
+  VERSION=$(git rev-parse --abbrev-ref HEAD | tr / -)-$(date +%Y-%m-%d)-$(git rev-parse --short HEAD)
+fi
+
+# build and push arch specific images
+for arch in ${CPU_ARCHS}; do
+  DOCKER_REPO="${QUAY_PATH}" GOARCH="${arch}" make container
```
Comment (paulfantom) [inline scripts/publish.sh:22]: QUAY_PATH is not set in the script, but there is `set -u` on top of the script which means script should fail to execute.

File: scripts/publish.sh (hunk 2 above)
Comment (s-urbaniak) [reply scripts/publish.sh:12]: eh, it's copy&paste, we can indeed remove this.

File: scripts/publish.sh (hunk 3 above)
Comment (s-urbaniak) [reply scripts/publish.sh:22]: QUAY_PATH is injected by an environment variable from with CI, but we can set a default here as well.

**Comment by s-urbaniak** [comment]
@paulfantom addressed comments, ptal

**Review by paulfantom — APPROVED** [review]
lgtm

**Comment by s-urbaniak** [comment]
eh, we have some issues with `git describe` in github actions:
```
fatal: No names found, cannot describe anything.
```
looking into it.

**Pull request merged**